crates/ruff_linter/src/rules/flake8_class_newline/rules/missing_class_newline.rs

```diff
@@ -0,0 +1,115 @@
+use ruff_diagnostics::AlwaysFixableViolation;
+use ruff_diagnostics::Diagnostic;
+use ruff_diagnostics::Edit;
+use ruff_diagnostics::Fix;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_codegen::Stylist;
+use ruff_python_parser::lexer::LexResult;
+use ruff_source_file::Locator;
+
+use crate::line_width::IndentWidth;
+use crate::rules::pycodestyle::rules::{LinePreprocessor, LogicalLineInfo, LogicalLineKind};
+
+/// ## What it does
+/// Checks for a missing blank line between a class definition and its first method.
+///
+/// ## Why is this bad?
+/// PEP8 says we should surround every class method with a single blank line.
+/// However it is ambiguous about the first method having a blank line above it.
+/// This rule enforces a single blank line, for consistency across classes.
```

Comment:
I think we should exclude `pyi` files from this rule because empty lines in stub files should only be used for grouping methods.